python/meterpreter/ext_server_stdapi.py

```diff
@@ -580,6 +602,71 @@ def get_username_from_token(token_user):
 		return None
 	return str(ctypes.string_at(domain)) + '\\' + str(ctypes.string_at(user))
 
+def get_windll_lang():
+	if not hasattr(ctypes.windll.kernel32, 'GetSystemDefaultLangID'):
+		return None
+	kernel32 = ctypes.windll.kernel32
+	kernel32.GetSystemDefaultLangID.restype = ctypes.c_uint16
+	lang_id = kernel32.GetSystemDefaultLangID()
+
+	size = kernel32.GetLocaleInfoW(lang_id, LOCALE_SISO3166CTRYNAME, 0, 0)
+	ctry_name = (ctypes.c_wchar * size)()
+	kernel32.GetLocaleInfoW(lang_id, LOCALE_SISO3166CTRYNAME, ctry_name, size)
+
+	size = kernel32.GetLocaleInfoW(lang_id, LOCALE_SISO639LANGNAME, 0, 0)
+	lang_name = (ctypes.c_wchar * size)()
+	kernel32.GetLocaleInfoW(lang_id, LOCALE_SISO639LANGNAME, lang_name, size)
+
+	if not (len(ctry_name.value) and len(lang_name)):
+		return 'Unknown'
+	return lang_name.value + '_' + ctry_name.value
+
+def get_windll_os_name():
+	os_info = windll_RtlGetVersion()
+	if not os_info:
+		return None
+	is_workstation = os_info.wProductType == VER_NT_WORKSTATION
+	os_name = None
+	if os_info.dwMajorVersion == 3:
+		os_name = 'NT 3.51'
+	elif os_info.dwMajorVersion == 4:
+		if os_info.dwMinorVersion == 0 and os_info.dwPlatformId == VER_PLATFORM_WIN32_WINDOWS:
+			os_name = '95'
+		elif os_info.dwMinorVersion == 10:
+			os_name = '98'
+		elif os_info.dwMinorVersion == 90:
+			os_name = 'ME'
+		elif os_info.dwMinorVersion == 0 and os_info.dwPlatformId == VER_PLATFORM_WIN32_NT:
+			os_name = 'NT 4.0'
+	elif os_info.dwMajorVersion == 5:
+		if os_info.dwMinorVersion == 0:
+			os_name = '2000'
+		elif os_info.dwMinorVersion == 1:
+			os_name = 'XP'
+		elif os_info.dwMinorVersion == 2:
+			os_name = '.NET Server'
+	elif os_info.dwMajorVersion == 6:
+		if os_info.dwMinorVersion == 0:
+			os_name = ('Vista' if is_workstation else '2008')
+		elif os_info.dwMinorVersion == 1:
+			os_name = ('7' if is_workstation else '2008 R2')
+		elif os_info.dwMinorVersion == 2:
+			os_name = ('8' if is_workstation else '2012')
+		elif os_info.dwMinorVersion == 3:
+			os_name = ('8.1' if is_workstation else '2012 R2')
+	elif os_info.dwMajorVersion == 10:
+		if os_info.dwMinorVersion == 0:
+			os_name = ('10' if is_workstation else '2016 Tech Preview')
```

Comment:
isn't 2016 officially out as of mid October? I seem to be able to download it from MSDN at any rate